packages/Autocomplete/src/index.tsx

```diff
@@ -0,0 +1,84 @@
+import Input from "@blaze-react/input";
+import utils from '@blaze-react/utils';
+import React, { Fragment, useState } from "react";
+
+interface IAutocompleteProps {
+  data: {
+    keyValue: string;
+    filterBy: any[];
+    data: object[];
+  };
+  utils: {
+    uniqueId: (element: any) => string
+  };
+  selected: (...args: any[]) => any;
+  placeholder?: string;
+  children?: any;
+}
+const Autocomplete: React.SFC<IAutocompleteProps> = ({
+  data: { data, filterBy: keys, keyValue },
+  placeholder,
+  selected,
+  utils: {
+    uniqueId
+  },
+  children
+}) => {
+  const [inputValue, setInputValue] = useState('');
+  const [showSelect, setShowSelect] = useState(false);
+
+  const handleChange = ({ value }: { value: string }) => {
+    setShowSelect(true);
+    setInputValue(value)
+    filterByValue(value);
+  };
+
+  const filterByValue = (value: string) => {
+    return data.filter(copy => {
+      let isMatch = false;
+      keys.forEach(key => {
```

Comment:
Could why try to simplify this code in order to don't have all this ifs, maybe with filter?

Reply:
done